lib/screens/search/search_page.dart

```diff
@@ -0,0 +1,135 @@
+      final query = ref.read(queryProvider.notifier).state;
+      await ref.read(searchViewModelProvider.notifier).initialize(query: query);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(child: _buildBody()),
+      backgroundColor: const Color(AppColors.background),
+    );
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  Widget _buildBody() {
+    final hideItemsOnView =
+        ref.read(searchViewModelProvider.notifier).hideItemsOnView;
+    final title = ref.read(searchViewModelProvider.notifier).title;
+
+    return ref.watch(
+      searchViewModelProvider.select(
+        (viewModel) => viewModel.when(
+          loading: () => const Center(
+            child: CircularProgressIndicator(
+              color: Color(AppColors.outline),
+              semanticsValue: 'Loading',
+            ),
+          ),
+          success: (list, noResults) => CustomScrollView(slivers: <Widget>[
+            SliverAppBar(
+              leading: !hideItemsOnView
+                  ? null
+                  : IconButton(
+                      icon: const Icon(Icons.arrow_back, color: Colors.black),
+                      onPressed: () {
+                        Navigator.of(context).pop();
+                        ref.read(queryProvider.notifier).state = null;
+                      },
+                    ),
+              backgroundColor: Colors.white,
+              floating: true,
+              pinned: true,
+              expandedHeight: 100,
+              title: hideItemsOnView
+                  ? Text(
+                      title,
+                      style: const TextStyle(color: Colors.black),
+                    )
+                  : null,
+              flexibleSpace: hideItemsOnView
+                  ? null
+                  : Padding(
+                      padding: const EdgeInsets.all(
+                        10,
+                      ),
+                      child: TextField(
+                        decoration: const InputDecoration(
+                            hintText: 'Search for cocktails'),
+                        onSubmitted: (String value) async {
+                          await ref
+                              .read(searchViewModelProvider.notifier)
+                              .search(query: value);
+                          ref.read(queryProvider.notifier).state = value;
+                        },
+                      ),
+                    ),
+            ),
+            SliverList(
+              delegate: SliverChildBuilderDelegate(
+                (BuildContext context, int index) {
+                  return GenericCard(
+                    url: list[index].img,
+                    name: list[index].name,
+                    outlineImg: false,
+                    onTap: () {
+                      ref.read(cocktailProvider.notifier).state = list[index];
+                      Navigator.push(
```

Comment:
Please move all this widget to a new Widget calls definition so it is more readable